### ISSUE TITLE
Process.detach with invalid type raises wrong error type

### ISSUE THREAD

**Issue by djberg96** [oracle/truffleruby#2782]
truffleruby 22.3.0, like ruby 3.0.3, GraalVM CE Native [x86_64-darwin]

`Process.detach("bogus")`

Expected: "no implicit conversion of String into Integer (TypeError)"
Got: "comparison of String with 0 failed (ArgumentError)"

**Comment by andrykonchin** [comment]
Thank you for reporting!

I can reproduce the issue:

CRuby 3.1.2

```ruby
 Process.detach("100")
#(irb):26:in `detach': no implicit conversion of String into Integer (TypeError)
```

TruffleRuby
```ruby
Process.detach("100")
# <internal:core> core/comparable.rb:44:in `>': comparison of String with 0 failed (ArgumentError)
```

**Comment by andrykonchin** [comment]
Fixed in 2eebebcdd05d1b7149f2f2a137dc2304211f3a9c
